src/value.cpp

```diff
@@ -491,7 +491,9 @@ expr MemRef::getWellDefined() const {
 
 optional<tuple<vector<expr>, MemRef::Layout, z3::sort>>
 MemRef::getDimsAndLayoutAndElemTy(
-    mlir::MemRefType memRefTy, bool freshVarForUnknownSize) {
+    mlir::MemRefType memRefTy,
+    optional<vector<expr>> predefinedDims,
```

Comment:
Let's make this either `optional<vector<expr>>&&` or `const optional<vector<expr>>&`

Reply:
I think it's enough to pass argument with `std::move` in caller. With move, it doesn't introduce value copy.
https://github.com/aqjune/mlir-tv/pull/73/commits/2d2f9999c3827e7fd300e009052bf2ac1e7f1be7#diff-c12f316dc5036218e6ee70c5052677d9601717720f54845049bbe49a4a8acf89R435